processor/src/test/java/org/mapstruct/ap/test/targetthis/CustomerDTO.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+public class CustomerDTO {
+    private String name;
+    private int level;
+    private EntityDTO entity;
```

Comment:
it is a bit confusing to stick an class EntityDTO inside a customerDTO.. Wouldn't `StatusDto` be better? (you normally map a DTO to an entity)

Reply:
Entity here is just generic name for a thing, for practice purposes it's just to mimic hierarchical/flat structure:
Customer extends Entity {}; // when you have inheritance
CustomerDTO { EntityDTO }; // when you don't have inheritance.

This is a bit of mirroring our data model and there Entity is basic class that every other inherits (directly or via another parent). In some other models I have seen "Record", "Item" and similar things. On it's own Entity would be like abstract class as it doesn't hold any "real world" value, but contains properties that every other object must have (typically it's something like "created, updated, status, deleted etc").
StatusDTO would be like proper attribute for CustomerDTO, but IMHO it would be wrong to inherit from it, aka "Customer extends Status {}".

Reply:
> Entity here is just generic name for a thing, for practice purposes it's just to mimic hierarchical/flat structure:
> Customer extends Entity {}; // when you have inheritance
> CustomerDTO { EntityDTO }; // when you don't have inheritance.

i understood as much.. However, here the purpose is to make clear that you map A to B. If the same "terminology"  is used in A and B this will be confusing. I guess that many developers automatically associate entity with a JPA entity (although that is not your intention) and DTO with an object that should be converted in an entity (and visa versa).




> this is a bit of mirroring our data model and there Entity is basic class that every other inherits (directly or via another parent).

Same here.. In my line of business we also talk about generic entities and attributes... So I can understand were you're coming from.. But its not about me only :smile:

Reply:
We aren't using database in that part of a project, so I totally forgot about database, but when tried to look with a bit of "fresh" eyes, first thing that came when reading entity was JPA. 
I changed everywhere to "item", hopefully it still logical (Customer extends Item) but with less confusion.